test/jetstream/api/kv_test.exs

```diff
@@ -66,6 +66,34 @@ defmodule Jetstream.API.KVTest do
     assert :ok = KV.delete_bucket(:gnat, "KEY_PUT_TEST")
   end
 
+  describe "watch/3" do
+    setup do
+      bucket = "KEY_WATCH_TEST"
+      {:ok, _} = KV.create_bucket(:gnat, bucket)
+      %{bucket: bucket}
+    end
+
+    test "detects key added and removed keys", %{bucket: bucket} do
+      {:ok, watchres} =
+        KV.watch(:gnat, bucket, fn action, key, value ->
+          # turn this into an assert?
+          IO.inspect(action)
```

Comment:
you'll want to leverage [assert_receive](https://hexdocs.pm/ex_unit/1.12/ExUnit.Assertions.html#assert_receive/3), but I think you'll want to do it outside of this function. `assert_receive` will wait 100ms by default for a message to the test process. Before the `watch` invocation you can do `test_pid = self()`. Inside this function you can do `send(test_pid, {action, key, value})`. This way you can send a message to the test process anytime the handler is invoked. Then after the `watch` invocation do 
```
assert_receive({:key_added, "foo", "bar})
assert_receive({:key_added, "baz", "quz"})
assert_receive({:key_deleted, "baz", "quz"})